Write (:w) in HTML files selects the whole file and loses position when HTML LSP is active

### Summary

Issuing a `:w` command in an HTML file selects the whole file and loses your current position when HTML LSP is active.

### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. `npm i -g vscode-langservers-extracted` to install HTML LSP
1. Create HTML file
2. `:w`

## I expected this to happen:

File is saved. Nothing is selected. My position is saved.

## Instead, this happened:

File is saved. Everything is selected. My position is lost (reverts to top).

Note that this behaviour does _not_ occur if the LSP is not installed/active.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2022-09-16T11:57:07.545 helix_term::application [ERROR] Language Server: Method not found client/r
```

</details>


### Platform

Linux (Fedora Silverblue 36)

### Terminal Emulator

Black Box 

### Helix Version

helix 22.08.1 (714db9c6)

--- a/helix-lsp/src/lib.rs
+++ b/helix-lsp/src/lib.rs
@@ -204,6 +204,20 @@ pub mod util {
         // in reverse order.
         edits.sort_unstable_by_key(|edit| edit.range.start);
 
+        // Generate a diff if the edit is a full document replacement.
+        #[allow(clippy::collapsible_if)]
+        if edits.len() == 1 {
+            let is_document_replacement = matches!(edits.first().and_then(|edit| {
+                let start = lsp_pos_to_pos(doc, edit.range.start, offset_encoding)?;
+                let end = lsp_pos_to_pos(doc, edit.range.end, offset_encoding)?;
+                Some(start..end)
+            }), Some(range) if range == (0..doc.len_chars()));
+            if is_document_replacement {
+                let new_text = Rope::from(edits.pop().unwrap().new_text);
+                return helix_core::diff::compare_ropes(doc, &new_text);
+            }
+        }
+
         Transaction::change(
             doc,
             edits.into_iter().map(|edit| {

--- a/helix-lsp/src/lib.rs
+++ b/helix-lsp/src/lib.rs
@@ -207,11 +207,11 @@ pub mod util {
         // Generate a diff if the edit is a full document replacement.
         #[allow(clippy::collapsible_if)]
         if edits.len() == 1 {
-            let is_document_replacement = matches!(edits.first().and_then(|edit| {
+            let is_document_replacement = edits.first().and_then(|edit| {
                 let start = lsp_pos_to_pos(doc, edit.range.start, offset_encoding)?;
                 let end = lsp_pos_to_pos(doc, edit.range.end, offset_encoding)?;
                 Some(start..end)
-            }), Some(range) if range == (0..doc.len_chars()));
+            }) == Some(0..doc.len_chars());
             if is_document_replacement {
                 let new_text = Rope::from(edits.pop().unwrap().new_text);
                 return helix_core::diff::compare_ropes(doc, &new_text);

